core/trino-main/src/main/java/io/trino/sql/planner/TranslationMap.java

```diff
@@ -674,20 +672,12 @@ private io.trino.sql.ir.Expression translate(DereferenceExpression expression)
 
     private io.trino.sql.ir.Expression translate(Array expression)
     {
-        checkCondition(expression.getValues().size() <= 254, TOO_MANY_ARGUMENTS, "Too many arguments for array constructor");
-
-        List<Type> types = expression.getValues().stream()
-                .map(analysis::getType)
-                .collect(toImmutableList());
-
         List<io.trino.sql.ir.Expression> values = expression.getValues().stream()
                 .map(this::translateExpression)
                 .collect(toImmutableList());
 
-        return BuiltinFunctionCallBuilder.resolve(plannerContext.getMetadata())
-                .setName(ArrayConstructor.NAME)
-                .setArguments(types, values)
-                .build();
+        Type type = analysis.getType(expression);
+        return new io.trino.sql.ir.Array(((ArrayType) type).getElementType(), values);
```

Comment:
Can this be imported?

Reply:
No, it conflicts with Array from the AST